Add 2D Parallax documentation page (#9587)
godotengine/godot-docs: tutorials/2d/2d_parallax.rst

.. doc_2d_parallax:

2D Parallax
===========

Introduction
------------

Parallax is an effect used to simulate depth by having textures move at different speeds relative to the camera. Godot

> provides the :ref:`Parallax2D<class_parallax2d>` node to make this easier for users. It can still be easy to get tripped
> up though, so this page provides in-depth descriptions of some properties and how to fix some common mistakes.

Reviewer: ```suggestion
provides the :ref:`Parallax2D<class_parallax2d>` node to achieve this effect. This node is still experimental, so the
implementation might change in future versions of Godot. However it is still recommended to use this over the old method, using the
:ref:`ParallaxLayer<class_parallaxlayer>` and :ref:`ParallaxBackground<class_parallaxbackground>` nodes. As such this page
will only cover how to use :ref:`Parallax2D<class_parallax2d>`.
```

.. doc_2d_parallax:

2D Parallax
===========

Introduction
------------

> Parallax is an effect used to simulate depth by having textures move at different speeds relative to the camera. It's
> also something that even experienced users struggle to wrap their heads around. Godot provides the

Reviewer 2: ```suggestion
Parallax is an effect used to simulate depth by having textures move at different speeds relative to the camera. Godot provides the
```
IMO that does not add much :)

:ref:`repeat_size<class_parallax2d_property_repeat_size>` tells the node to snap its position forward or back when the
camera scrolls by the set value. This effect is achieved by adding a single repeat to all the child canvas items offset
by the value. While the camera scrolls between the image and its repeat, it invisibly snaps back giving the appearance
of a looping image.

.. image:: img/2d_parallax_scroll.gif

Being a delicate effect, it's easy for unfamiliar users to make mistakes with their setup. Let's go over the "how" and
"why" of a few common problems users encounter.

Poor sizing
~~~~~~~~~~~

The infinite repeat effect is easiest to work with when you have an image designed to repeat seamlessly and is the same
size or larger than your viewport **before** setting the :ref:`repeat_size<class_parallax2d_property_repeat_size>`. If
you aren't able to obtain assets that are designed for this task, there are some other things you can do to better
prepare your image in regards to size.

Here is an example of a texture that is too small for its viewport:

.. image:: img/2d_parallax_size_bad.webp

> We can see that the viewport size is ``500 x 300`` but the texture is ``288 x 208``. If we set the

Reviewer 2: ```suggestion
We can see that the viewport size is 500×300 but the texture is 288×208. If we set the
```
Either no formatting, or alternatively just for the numbers. I think this is fine? Also unsure whether we should add "pixel", but this is probably clear enough as is.

:ref:`Parallax2D.repeat_size<class_parallax2d_property_repeat_size>` is set correctly, it should look something like
this, with one single loop of the image the same size or larger than the viewport:

.. image:: img/2d_parallax_repeat_good_norect.webp

If you think of how the image scrolls across the screen, it starts by displaying what's inside the red rectangle
(determined by :ref:`repeat_size<class_parallax2d_property_repeat_size>`), and when it reaches what's inside the yellow
rectangle it zips the image forward to give the illusion of scrolling forever.

.. image:: img/2d_parallax_repeat_good.webp

If you have the image positioned away from the "infinite repeat canvas", when the camera reaches the yellow rectangle,
half of the image is cut off before it jumps forward like in the image below:

.. image:: img/2d_parallax_repeat_bad.webp

Scroll offset
-------------

If your parallax textures are already working correctly, but you prefer it to start at a different point,
:ref:`Parallax2D<class_parallax2d>` comes with a :ref:`scroll_offset<class_parallax2d_property_scroll_offset>` property

> used to offset where the infinite repeat canvas starts. As an example, if your image is ``288 x 208``, setting

Reviewer 2: ```suggestion
used to offset where the infinite repeat canvas starts. As an example, if your image is 288×208, setting
```

the :ref:`scroll_offset<class_parallax2d_property_scroll_offset>` to ``(-144,0)`` or ``(144,0)`` allows it to begin
halfway across the image.

Repeat times
------------

Ideally, following this guide, your parallax textures are large enough to cover the screen even when zoomed out.

> Until now, we have had a perfectly fitting ``288 x 208`` texture inside of a ``288 x 208`` viewport. However, problems 

Reviewer 2: ```suggestion
Until now, we have had a perfectly fitting 288×208 texture inside of a 288×208 viewport. However, problems 
```
